gapis/shadertools/cc/staticanalysis.cpp

```diff
@@ -20,6 +20,90 @@
 #include "staticanalysis.h"
 
 #include <cstring>
+#include <map>
+#include <set>
+#include <vector>
+
+struct AnalysisBlock {
+  std::set<uint32_t> def;
+  std::set<uint32_t> use;
+  std::set<uint32_t> in;
+  std::set<uint32_t> out;
+  std::vector<uint32_t> successors;
+  std::map<uint32_t, uint32_t> firstMade;
+  std::map<uint32_t, uint32_t> lastUse;
+};
+
+bool isValidIDForPressure(spirv_cross::ParsedIR pir, uint32_t id,
+                          spirv_cross::Instruction op, uint32_t offset) {
+  if (pir.ids[id].get_type() == spirv_cross::TypeNone) {
+    switch (op.op) {
+      case spv::OpExtInst:
+        return offset != 3;
+
+      case spv::OpVectorShuffle:
+        return offset < 4;
+
+      case spv::OpArrayLength:
+        return offset != 3;
+
+      case spv::OpCompositeExtract:
+        return offset < 3;
+
+      case spv::OpCompositeInsert:
+        return offset < 4;
+
+      case spv::OpConstantPipeStorage:
```

Comment:
This instruction isnt allowed in Vulkan-flavored SPIRV at all. Did you see it in a shader?

Reply:
No. I was just going through the SPIRV spec quickly. I can remove it.